New Analog App (nx v17) fails to build - trpc issue?

### Please provide the environment you discovered this bug in.

I tried installing Analog using the NX preset on:

My Macbook and on gitpod.io - same results.

### Which area/package is the issue in?

create-analog

### Description

After creating a new NX Workspace with the Analog preset the `build` fails.

### Please provide the exception or error you saw

```true
$ npx create-nx-workspace@latest --preset=@analogjs/platform                                                                                                 

 >  NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

✔ Where would you like to create your workspace? · my-org
✔ Enable distributed caching to make your CI faster · Yes

 >  NX   Creating your v17.0.1 workspace.

   To make sure the command works reliably in all environments, and that the preset is applied correctly,
   Nx will run "npm install" several times. Please wait.

✔ Installing dependencies with npm
✔ Successfully created the workspace: my-org.
✔ What name would you like to use for your Analog app? · analog-app
✔ Add TailwindCSS for styling? (Y/n) · true
✔ Add tRPC for typesafe client/server interaction? (y/N) · true
Angular has not been installed yet. Creating an Angular application
Fetching @nx/angular...
Fetching @angular-devkit/core...
Fetching prettier...
✔ NxCloud has been set up successfully

 >  NX   Distributed caching via Nx Cloud has been enabled

   In addition to the caching, Nx Cloud provides config-free distributed execution,
   UI for viewing complex runs and GitHub integration. Learn more at https://nx.app
   
   Your workspace is currently unclaimed. Run details from unclaimed workspaces can be viewed on cloud.nx.app by anyone
   with the link. Claim your workspace at the following link to restrict access.
   
   https://cloud.nx.app/orgs/workspace-setup?accessToken=[my-access-token]


 >  NX   Successfully applied preset: @analogjs/platform

$ npx nx build analog-app                                                                                                                        

> nx run analog-app:build:production

node_modules/@trpc/client/src/TRPCClientError.ts:42:18 - error TS4111: Property 'error' comes from an index signature, so it must be accessed with ['error'].

  40 |   return (
  41 |     isObject(obj) &&
> 42 |     isObject(obj.error) &&
     |                  ^
  43 |     typeof obj.error.code === 'number' &&
  44 |     typeof obj.error.message === 'string'
  45 |   );

node_modules/@trpc/client/src/TRPCClientError.ts:43:16 - error TS4111: Property 'error' comes from an index signature, so it must be accessed with ['error'].

  41 |     isObject(obj) &&
  42 |     isObject(obj.error) &&
> 43 |     typeof obj.error.code === 'number' &&
     |                ^
  44 |     typeof obj.error.message === 'string'
  45 |   );
  46 | }

node_modules/@trpc/client/src/TRPCClientError.ts:43:22 - error TS4111: Property 'code' comes from an index signature, so it must be accessed with ['code'].

  41 |     isObject(obj) &&
  42 |     isObject(obj.error) &&
> 43 |     typeof obj.error.code === 'number' &&
     |                      ^
  44 |     typeof obj.error.message === 'string'
  45 |   );
  46 | }

node_modules/@trpc/client/src/TRPCClientError.ts:44:16 - error TS4111: Property 'error' comes from an index signature, so it must be accessed with ['error'].

  42 |     isObject(obj.error) &&
  43 |     typeof obj.error.code === 'number' &&
> 44 |     typeof obj.error.message === 'string'
     |                ^
  45 |   );
  46 | }
  47 | 

node_modules/@trpc/client/src/TRPCClientError.ts:44:22 - error TS4111: Property 'message' comes from an index signature, so it must be accessed with ['message'].

  42 |     isObject(obj.error) &&
  43 |     typeof obj.error.code === 'number' &&
> 44 |     typeof obj.error.message === 'string'
     |                      ^
  45 |   );
  46 | }
  47 | 

node_modules/@trpc/client/src/TRPCClientError.ts:103:7 - error TS2322: Type 'TRPCClientError<ErrorInferrable>' is not assignable to type 'TRPCClientError<TRouterOrProcedure>'.
  Type 'ErrorInferrable' is not assignable to type 'TRouterOrProcedure'.
    'ErrorInferrable' is assignable to the constraint of type 'TRouterOrProcedure', but 'TRouterOrProcedure' could be instantiated with a different subtype of constraint 'ErrorInferrable'.
      Type 'AnyProcedure' is not assignable to type 'TRouterOrProcedure'.
        'AnyProcedure' is assignable to the constraint of type 'TRouterOrProcedure', but 'TRouterOrProcedure' could be instantiated with a different subtype of constraint 'ErrorInferrable'.

  101 |     }
  102 |     if (isTRPCErrorResponse(cause)) {
> 103 |       return new TRPCClientError(cause.error.message, {
      |       ^
  104 |         ...opts,
  105 |         result: cause,
  106 |       });

Found 6 errors.

 >  NX   Found type errors. See above.

   Pass --verbose to see the stacktrace.


 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project analog-app (3s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```


### Other information

Adding the `--verbose` flag doesn't provide more information.

Setting `noPropertyAccessFromIndexSignature` to false in the tsconfig.json removes the TS4111-Errors.

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

--- a/packages/trpc/src/lib/client/client.ts
+++ b/packages/trpc/src/lib/client/client.ts
@@ -1,6 +1,11 @@
 import { InjectionToken, Provider, signal, TransferState } from '@angular/core';
 import 'isomorphic-fetch';
-import { httpBatchLink, HttpBatchLinkOptions } from '@trpc/client';
+import {
+  httpBatchLink,
+  HttpBatchLinkOptions,
+  CreateTRPCClientOptions,
+  HTTPHeaders,
+} from '@trpc/client';
 import { AnyRouter } from '@trpc/server';
 import { transferStateLink } from './links/transfer-state-link';
 import {
@@ -9,8 +14,6 @@ import {
   tRPC_CACHE_STATE,
 } from './cache-state';
 import { createTRPCRxJSProxyClient } from './trpc-rxjs-proxy';
-import { CreateTRPCClientOptions } from '@trpc/client/src/createTRPCUntypedClient';
-import { HTTPHeaders } from '@trpc/client/src/links/types';
 import { FetchEsque } from '@trpc/client/dist/internals/types';
 
 export type TrpcOptions<T extends AnyRouter> = {

--- a/packages/trpc/src/lib/client/shared-internal.ts
+++ b/packages/trpc/src/lib/client/shared-internal.ts
@@ -6,11 +6,12 @@ import {
   DefaultDataTransformer,
 } from '@trpc/server';
 import {
+  Operation,
+  TRPCLink,
   OperationContext,
   OperationLink,
   OperationResultObservable,
-} from '@trpc/client/src/links/types';
-import { Operation, TRPCLink } from '@trpc/client';
+} from '@trpc/client';
 import { observable } from '@trpc/server/observable';
 
 // Removed subscription

--- a/packages/trpc/src/lib/client/trpc-rxjs-proxy.ts
+++ b/packages/trpc/src/lib/client/trpc-rxjs-proxy.ts
@@ -17,21 +17,24 @@ import {
   createRecursiveProxy,
   inferTransformedProcedureOutput,
 } from '@trpc/server/shared';
-import { inferObservableValue, share } from '@trpc/server/observable';
 import {
+  inferObservableValue,
+  share,
+  Observable as TrpcObservable,
+} from '@trpc/server/observable';
+import { Observable as RxJSObservable } from 'rxjs';
+import {
+  TRPCClientError,
   OperationContext,
   OperationLink,
   TRPCClientRuntime,
-} from '@trpc/client/src/links/types';
-import { Observable as RxJSObservable } from 'rxjs';
-import { TRPCClientError } from '@trpc/client';
+} from '@trpc/client';
 import {
   createChain,
   CreateTRPCClientOptions,
   TRPCRequestOptions,
   TRPCType,
 } from './shared-internal';
-import { Observable as TrpcObservable } from '@trpc/server/src/observable/types';
 
 // Changed to rxjs observable
 type Resolver<TProcedure extends AnyProcedure> = (

